### PR TITLE
143 refactor creating requests

### PR DESCRIPTION
**Pull request by alishaevn**
# Story
I noticed some areas for improvement that may also help our future selves while working on #226 

# Expected Behavior After Changes
- simplify `posting` so it is easier to use in other use cases in the future
- change the prop name passed into `createRequest` from "data" to "dynamicFormData" to remove conflicts after updating `posting`
- only try to attach files to a new request if there are any. also, account for attachment errors
- account for errors on the new request page and refactor for the new return values from `posting`
- an empty object is still truthy. check that the error has data in `configureErrors`
- checking that an array has length instead of if the length does not equal 0 (same thing, little less code)

# Screenshots / Video

<details>
<summary></summary>

</details>

# Notes

### CODE REVIEW

File: utils/api/base.js [hunk 1]
```diff
@@ -18,15 +18,10 @@ export const fetcher = (url, token) => {
...
-      return {
-        success: true,
-        error: false,
-        requestID,
-        quotedWareID,
-      }
+
+    return {
+      data: response.data,
+      error: false,
```
Comment (alishaevn) [inline utils/api/base.js:24]: simplify `posting` so it is easier to use in other use cases in the future. also, it shouldn't care about the data so much. the function calling it should.

File: utils/api/configurations.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ export const configureRequests = ({ data, path }) => {
 export const configureErrors = (errors) => {
   const env = process.env.NODE_ENV
   const remainingErrors = errors
-    .filter(error => error)
+    .filter(error => Object.keys(error).length)
```
Comment (alishaevn) [inline utils/api/configurations.js:72]: an empty object is still truthy. 😅

File: utils/api/configurations.js [hunk 3]
```diff
@@ -161,7 +161,7 @@ export const configureMessages = (data) => {
 
 export const  configureFiles = (data) => {
   // filter out the notes that do not have attachments
-  const notesWithFiles = data.filter(d => d.attachments?.length !== 0)
+  const notesWithFiles = data.filter(d => d.attachments?.length)
```
Comment (alishaevn) [inline utils/api/configurations.js:164]: same result, fewer characters.

File: utils/api/requests.js [hunk 4]
```diff
@@ -139,60 +139,72 @@ export const createMessageOrFile = ({ id, quotedWareID, message, files, accessTo
...
-      state: data.shipping.state,
-      street: data.shipping.street,
-      street2: data.shipping.street2,
-      zipcode: data.shipping.zipcode,
+      city: dynamicFormData.shipping.city,
+      country: dynamicFormData.shipping.country,
+      state: dynamicFormData.shipping.state,
+      street: dynamicFormData.shipping.street,
+      street2: dynamicFormData.shipping.street2,
+      zipcode: dynamicFormData.shipping.zipcode,
```
Comment (alishaevn) [inline utils/api/requests.js:188]: changed these to "dynamicFormData" so they don't conflict with the new "data" variable on line 193.

File: utils/api/requests.js [hunk 5]
```diff
@@ -139,60 +139,72 @@ export const createMessageOrFile = ({ id, quotedWareID, message, files, accessTo
...
       organization_name: process.env.NEXT_PUBLIC_PROVIDER_NAME
     },
   }
 
-  const response = await posting(`/wares/${wareID}/quote_groups.json`, { pg_quote_group }, accessToken)
-  createMessageOrFile({ id: response.requestID, files: data.attachments, quotedWareID: response.quotedWareID })
+  let { data, error } = await posting(`/wares/${wareID}/quote_groups.json`, { pg_quote_group }, accessToken)
 
-  return response
+  if (data && dynamicFormData.attachments) {
```
Comment (alishaevn) [inline utils/api/requests.js:195]: no need trying to add attachments if we don't have any. save time and an api call.

File: utils/api/base.js (hunk 1 above)
Comment (alishaevn) [reply utils/api/base.js:24]: this also aligns our return values with the return values from `useSWR`. 😎

**Review by summer-cook — APPROVED** [review]

**Pull request merged**